docs/guides/debugging.md

```diff
@@ -0,0 +1,91 @@
+# Debugging
+
+## Debugging? What it is and why?
+
+Debugging is one of **the most important things on C**, not only Concord. It is important **because it shows where the error** (AKA segmentation fault or core dump) **came from** since you cannot know it with ease without it.
```

Comment:
"Debugging is one of **the most important things** when it comes to C programming, not only Concord. It is important **because a debugger will show you where the error** (like, a segmentation fault or core dump) **came from**, since you cannot know it with ease without it."